### PR TITLE
Assert maxTotalChunkedDataInBytes when only chunked migration enabled

### PR DESCRIPTION
**Pull request by ahmetmircik**
PR should fix failing compatibility tests.

**Modification:**
Assertion will be active when chunked migration is enabled.

**Note:**
I ran compatibility tests and they are passing with this PR: http://jenkins.hazelcast.com/job/ahmet-Hazelcast-EE-pr-builder/190/

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/internal/partition/ChunkSerDeHelper.java [hunk 1]
```diff
@@ -46,10 +46,14 @@
...
     public ChunkSerDeHelper(ILogger logger, int partitionId,
                             Collection<ChunkSupplier> chunkSuppliers,
+                            boolean chunkedMigrationEnabled,
                             int maxTotalChunkedDataInBytes) {
         assert chunkSuppliers != null;
         assert logger != null;
-        assert maxTotalChunkedDataInBytes > 0 : maxTotalChunkedDataInBytes;
+
+        if (chunkedMigrationEnabled) {
+            assert maxTotalChunkedDataInBytes > 0 : maxTotalChunkedDataInBytes;
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/internal/partition/ChunkSerDeHelper.java:55]: maybe `assert !chunkedMigrationEnabled || (maxTotalChunkedDataInBytes > 0) : maxTotalChunkedDataInBytes;` to eliminate the `if` statement?

**Review by vbekiaris — APPROVED** [review]
added a minor suggestion

File: hazelcast/src/main/java/com/hazelcast/internal/partition/ChunkSerDeHelper.java (hunk 1 above)
Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/internal/partition/ChunkSerDeHelper.java]: done

**Comment by ahmetmircik** [comment]
Simple fix merging with 1 ok.

**Pull request merged**